src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -199,6 +199,35 @@ private VSTestResult GetBasicResult(XmlNode resultNode)
             if (outputNode != null)
                 vsResult.Messages.Add(new TestResultMessage(TestResultMessage.StandardOutCategory, outputNode.InnerText));
 
+            var attachments = resultNode.SelectNodes("attachments/attachment");
+            if (attachments.Count > 0)
+            {
+                var attachmentSet = new AttachmentSet(new Uri(NUnitTestAdapter.ExecutorUri), "Attachments");
+                vsResult.Attachments.Add(attachmentSet);
+
+                foreach (XmlNode attachment in attachments)
+                {
+                    var path = attachment.SelectSingleNode("filePath")?.InnerText ?? string.Empty;
+                    var description = attachment.SelectSingleNode("description")?.InnerText;
+
+                    try
+                    {
+                        // We only support absolute paths since we dont lookup working directory here
+                        // any problem with path will throw an exception
+                        var fileUri = new Uri(path, UriKind.Absolute);
+                        attachmentSet.Attachments.Add(new UriDataAttachment(fileUri, description));
+                    }
+                    catch (UriFormatException ex)
+                    {
+                        _logger.Warning(string.Format("Ignoring attachment with path '{0}' due to problem with path: {1}", path, ex.Message));
```

Comment:
VS2017 is free and once the SDK teams figure an architecture out, MSBuild and the SDKs (which are all open source) will be able to be bin deployed from NuGet rather than installed and used directly a la Cake without any version of VS installed at all. C# 7 is freely available to anyone in VS2015 as well if we are willing to reference the roslyn compilers as design-time project NuGet dependencies.